src/Language/AST/FieldDefinitionNode.php

```diff
@@ -1,35 +1,31 @@
 <?php
+
+declare(strict_types=1);
+
 namespace GraphQL\Language\AST;
 
 class FieldDefinitionNode extends Node
 {
-    /**
-     * @var string
-     */
+    /** @var string */
     public $kind = NodeKind::FIELD_DEFINITION;
 
-    /**
-     * @var NameNode
-     */
+    /** @var NameNode */
     public $name;
 
-    /**
-     * @var InputValueDefinitionNode[]|NodeList
-     */
+    /** @var InputValueDefinitionNode[]|NodeList */
     public $arguments;
 
-    /**
-     * @var TypeNode
-     */
+    /** @var TypeNode */
     public $type;
 
-    /**
-     * @var DirectiveNode[]|NodeList
-     */
+    /** @var DirectiveNode[]|NodeList */
```

Comment:
Shouldn't those be changed around (as well as in other places where NodeList occurs), because the property actually IS a NodeList

Reply:
Could you be more specific please? Not sure what you mean by that. Thanks

Reply:
I believe that all lists of related nodes in the AST are actually of the type `NodeList`, so the type annotation should match that. It is fine to have the underlying type of the NodeList as a secondary hint, so it should be `NodeList|DirectiveNode[]`

Reply:
Ah, I didn't touch this TBH. Only reformatted the annotation to be CS compliant.

The order of union type hints doesn't matter and sometimes can be relative. We have a CS check for having `null` as the last type but as long as there's no custom CS check for having NodeList as the first type, I don't really see a point in it.

I'd suggest you open a issue / PR with union types proper ordering if you want to pursue this (inspiration https://github.com/slevomat/coding-standard/blob/master/SlevomatCodingStandard/Sniffs/TypeHints/NullTypeHintOnLastPositionSniff.php).

Reply:
I am not actually sure that having both is actually correct. I just looked up the meaning of the pipe character `|` in types. In this case, it actually always is one exact type, `NodeList`.

Reply:
AFAIK pipe means `either a or b` but haven't checked whether both are correct or just one.

Reply:
Agreed. Go ahead then